service/tcp.go

```diff
@@ -285,19 +286,27 @@ func (s *tcpService) handleConnection(listenerPort int, clientTCPConn *net.TCPCo
 			tgtConn.CloseWrite()
 			fromClientErrCh <- fromClientErr
 		}()
-		_, fromTargetErr := ssw.ReadFrom(tgtConn)
```

Comment:
Are you saying that in some cases this never returns, even if `tgtConn` times out?
In that case, wouldn't the same happen to your go routine?

Reply:
I also don't understand the deadlock argument. The main go routine is waiting on the side one, but not the other way around, as far as I can tell.